clients/clearly-defined/src/main/kotlin/Enums.kt

```diff
@@ -43,7 +43,8 @@ enum class ComponentType(val defaultProvider: Provider? = null) {
 
     companion object {
         @JvmStatic
-        fun fromString(value: String) = ComponentType.entries.single { it.toString() == value }
+        fun fromString(value: String) =
+            ComponentType.entries[ComponentType.serializer().descriptor.getElementIndex(value)]
```

Comment:
Why not use `ComponentType.valueOf(value)` (and maybe remove this function altogether as it seems to do the same as `valueOf`)?

Reply:
`valueOf()` expects `ComponentType.name` as the `value`, but here we want to construct the enum from the name used for serialization (see `@SerialName`).